File: hencoderplus/src/main/java/com/example/hencoderplus/kotlin/CodeView.java
package com.example.hencoderplus.kotlin;

import android.content.Context;
import android.graphics.Canvas;
import android.graphics.Paint;
import android.util.AttributeSet;
import android.view.View;
import android.widget.TextView;

import androidx.annotation.Nullable;
import androidx.appcompat.widget.AppCompatTextView;

import com.example.hencoderplus.R;

import java.util.ArrayList;
import java.util.List;
import java.util.Random;

public class CodeView extends AppCompatTextView {

    private Paint paint;
    public  final String T1 = "FLASH";
    public  final String T2 = "SONIC";
    private List<String> list = new ArrayList<>();


    public CodeView(Context context, @Nullable AttributeSet attrs) {
        super(context, attrs);
    }

    {
        paint = new Paint(Paint.ANTI_ALIAS_FLAG);
        paint.setColor(getResources().getColor(R.color.colorAccent));
        paint.setStyle(Paint.Style.STROKE);
        paint.setStrokeWidth(20);
        paint.setStrokeCap(Paint.Cap.ROUND);

        list.add(T1);
        list.add(T2);
    }

    @Override
    protected void onDraw(Canvas canvas) {
        super.onDraw(canvas);
        canvas.drawLine(0,0,getWidth(),getHeight(),paint);
    }

    private void updateCode(){
        int index = (int) (Math.random() * list.size());
        setText(list.get(index));
        postInvalidate();
    }
}

File: hencoderplus/src/main/java/com/example/hencoderplus/kotlin/CodeView.kt
package com.example.hencoderplus.kotlin


import android.content.Context
import android.graphics.Canvas
import android.graphics.Paint
import android.util.AttributeSet
import androidx.appcompat.widget.AppCompatTextView
import com.example.hencoderplus.R

class CodeView : AppCompatTextView {

    private var paint: Paint = Paint(Paint.ANTI_ALIAS_FLAG)
    private val t1: String = "FLASH"
    private val t2: String = "SONIC"
    private val t3: String = "FLAME"
    private val t4: String = "STORM"
    private val t5: String = "THUNDER"
    private var temp: Int = 0
    private var list: List<String> = listOf(t1, t2, t3, t4, t5)

    init {
        paint.color = resources.getColor(R.color.colorAccent)
        paint.style = Paint.Style.STROKE
        paint.strokeWidth = 13F
        paint.strokeCap = Paint.Cap.ROUND

    }

    constructor(context: Context?) : this(context, null)

    constructor(context: Context?, attrs: AttributeSet?) : super(context, attrs)


    override fun onDraw(canvas: Canvas?) {
        super.onDraw(canvas)
        canvas?.drawLine(0F, 0F, width.toFloat(), height.toFloat(), paint)
    }

    fun updateCode() {
        var index: Int
        while (true) {
            index = (Math.random() * list.size).toInt()
            if (temp != index)
                break
        }
        temp = index
        val text: String = list[index]
        setText(text)
        postInvalidate()

    }
}
